lib/compat/wordpress-6.0/class-gutenberg-rest-global-styles-controller.php

```diff
@@ -0,0 +1,96 @@
+<?php
+/**
+ * REST API: Gutenberg_REST_Global_Styles_Controller class
+ *
+ * @package    Gutenberg
+ * @subpackage REST_API
+ */
+
+/**
+ * Base Global Styles REST API Controller.
+ */
+class Gutenberg_REST_Global_Styles_Controller extends WP_REST_Global_Styles_Controller {
+	/**
+	 * Registers the controllers routes.
+	 *
+	 * @return void
+	 */
+	public function register_routes() {
+		// List themes global styles.
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/themes/(?P<stylesheet>[\/\s%\w\.\(\)\[\]\@_\-]+)/variations',
+			array(
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_theme_items' ),
```

Comment:
I think we should rename `get_theme_items` to `get_theme_variations` or change the endpoint to be `theme-items`. This all comes down to whether we want to support more things from this endpoint other than fetching the variations.

Reply:
We will merge this into `WP_REST_Global_Styles_Controller` in the future, right? In that case using `get_theme_variations` or `get_variation_items` make sense.